R/derive_vars_transposed.R

```diff
@@ -84,19 +93,46 @@ derive_vars_transposed <- function(dataset,
                                    by_vars,
                                    key_var,
                                    value_var,
-                                   filter = NULL) {
+                                   filter = NULL,
+                                   relationship = NULL) {
   key_var <- assert_symbol(enexpr(key_var))
   value_var <- assert_symbol(enexpr(value_var))
   filter <- assert_filter_cond(enexpr(filter), optional = TRUE)
   assert_vars(by_vars)
   assert_data_frame(dataset, required_vars = replace_values_by_names(by_vars))
   assert_data_frame(dataset_merge, required_vars = expr_c(by_vars, key_var, value_var))
+  relationship <- assert_character_scalar(
+    relationship,
+    values = c("one-to-one", "many-to-one"),
+    case_sensitive = TRUE,
+    optional = TRUE
+  )
 
   dataset_transposed <- dataset_merge %>%
     filter_if(filter) %>%
     pivot_wider(names_from = !!key_var, values_from = !!value_var)
 
-  left_join(dataset, dataset_transposed, by = vars2chr(by_vars))
+  tryCatch(
+    left_join(dataset, dataset_transposed, by = vars2chr(by_vars), relationship = relationship),
+    "dplyr_error_join_relationship_one_to_one" = function(cnd) {
+      cli_abort(
+        message = paste0(
+          "Each row in `dataset` must match at most 1 row ",
+          "in the transposed `dataset_merge`."
```

Comment:
I think we should be more precise here. There are two cases which cause the error:
- A row in `dataset` matches multiple rows in `dataset_merge`.
- A row in `dataset_merge` matches multiple rows in `dataset`.

`left_join()` issues different error messages depending on the case. We should take this into account. Maybe we could just take the error message from `left_join()` and replace `x` with `dataset` and `y` with "the transposed `dataset_merge`".

Reply:
Finally figured how to do this! Hopefully that should work better now.